File: server/app/schemas/message.py
from datetime import datetime

from pydantic import BaseModel


class MessageResponse(BaseModel):
    id: int
    message_id: int
    chat_id: int
    chat_name: str | None
    sender_id: int | None
    sender_name: str | None
    text: str | None
    links: list[str] | None
    raw_date: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: web/src/types/message.ts
export interface MessageResponse {
  id: number
  message_id: number
  chat_id: number
  chat_name: string | null
  sender_id: number | null
  sender_name: string | null
  text: string | null
  links: string[] | null
  raw_date: string | null
  created_at: string
}

export interface MessageFilters {
  chat_id?: number
  skip?: number
  limit?: number
}
